### ISSUE TITLE
this framework what time support ios 11.0 and build in xcode 9.x bate

### ISSUE THREAD

**Issue by anker-eric** [CocoaLumberjack/CocoaLumberjack#892]
### New Issue Checklist
this framework what time support ios 11.0 and build in xcode 9.x bate.

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](https://github.com/CocoaLumberjack/CocoaLumberjack#documentation) or [CocoaDocs](cocoadocs.org/docsets/CocoaLumberjack)
* [ ] I have searched for a similar issue in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / osx / tvos / watchos
 Platform Version        | e.g. 8.0
 CocoaLumberjack Version | e.g. 2.3.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by kedu** [comment]
用xcdoe9运行就报错, if (message) {
        const char *msg = [message UTF8String];
        
        switch (logMessage->_flag) {
            case DDLogFlagError     :
                os_log_error(OS_LOG_DEFAULT, msg);
                break;
            case DDLogFlagWarning   :
            case DDLogFlagInfo      :
                os_log_info(OS_LOG_DEFAULT, msg);
                break;
            case DDLogFlagDebug     :
            case DDLogFlagVerbose   :
            default                 :
                os_log_debug(OS_LOG_DEFAULT, msg);
                break;
        }

//报错
Static_assert failed "format argument must be a string constant"

**Comment by Martell1994** [comment]
@kedu pod update CocoaLumberjack

**Comment by idbeny** [comment]
perform 'pod update CocoaLumberjack' later it's still repear same error

**Comment by prsolucoes** [comment]
I have the same problem on the code:

```
switch (logMessage->_flag) {
        case DDLogFlagError     :
            os_log_error(OS_LOG_DEFAULT, msg);
            break;
        case DDLogFlagWarning   :
        case DDLogFlagInfo      :
            os_log_info(OS_LOG_DEFAULT, msg);
            break;
        case DDLogFlagDebug     :
        case DDLogFlagVerbose   :
        default                 :
            os_log_debug(OS_LOG_DEFAULT, msg);
            break;
    }
//报错
Static_assert failed "format argument must be a string constant"
```

**Comment by animaonline** [comment]
Needs attention

**Comment by rivera-ernesto** [comment]
It should now with 3.3.0.

**Comment by sushichop** [comment]
This issue already should be fixed with CocoaLumberjack 3.3.0 as @rivera-ernesto says.

@bpoplauschi @rivera-ernesto
I think we can close this issue.
